manageiq-operator/pkg/helpers/miq-components/httpd.go

```diff
@@ -0,0 +1,443 @@
+						},
+					},
+					ServiceAccountName: cr.Spec.AppName+"-httpd",
+
+
+					
+				},
+			},
+		},
+	}
+}
+
+
+
+
+
+
+
+func NewUIService(cr *miqv1alpha1.Manageiq) *corev1.Service{
+	labels := map[string]string{
+		"app": cr.Spec.AppName,   
+	}
+	selector := map[string]string{
+		"service": "ui",
+	}
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "ui",
+			Namespace: cr.ObjectMeta.Namespace,
+			Labels: labels, 
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				corev1.ServicePort{
+					Name: "web-service-3000",
+					Port: 3000,
+				},
+			},
+			Selector: selector,
+		},
+	}
+}
+
+
+
+func NewWebService(cr *miqv1alpha1.Manageiq) *corev1.Service{
+	labels := map[string]string{
+		"app": cr.Spec.AppName,  
+	}
+	selector := map[string]string{
+		"service": "web-service",
+	}
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "web-service",
+			Namespace: cr.ObjectMeta.Namespace,
+			Labels: labels, 
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				corev1.ServicePort{
+					Name: "web-service-3000",
+					Port: 3000,
+				},
+			},
+			Selector: selector,
+		},
+	}
+}
+
+
+
+
+func NewRemoteConsoleService(cr *miqv1alpha1.Manageiq) *corev1.Service{
+	labels := map[string]string{
+		"app": cr.Spec.AppName,    
+	}
+	selector := map[string]string{
+		"service": "remote-console",    
+	}
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "remote-console",
+			Namespace: cr.ObjectMeta.Namespace,
+			Labels: labels, 
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				corev1.ServicePort{	
+					Name: "remote-console-3000",
+					Port: 80,
+				},
+			},
+			Selector: selector,	
+		},
+	}
+}
+
+
+
+
+func NewHttpdService(cr *miqv1alpha1.Manageiq) *corev1.Service{
+	labels := map[string]string{
+		"app": cr.Spec.AppName,   
+	}
+
+	selector := map[string]string{
+		"name": "httpd",
+	}
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "httpd",
+			Namespace: cr.ObjectMeta.Namespace,
+			Labels: labels, 
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				corev1.ServicePort {
+					Name: "http",
+					Port: 80,
+				},
+			},
+			Selector: selector,
+		},
+	}
+}
+
+func NewHttpdDbusAPIService(cr *miqv1alpha1.Manageiq) *corev1.Service{
+	labels := map[string]string{
+		"app": cr.Spec.AppName,   
+	}
+
+	selector := map[string]string{
+		"name": "httpd",
+	}
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "httpd-dbus-api",
+			Namespace: cr.ObjectMeta.Namespace,
+			Labels: labels, 
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				corev1.ServicePort {
+					Name: "http-dbus-api",
+					Port: 8080,
+				},
+			},
+			Selector: selector,
+		},
+	}
+}
+
+
+
+
+
+func TLSSecret(cr *miqv1alpha1.Manageiq) *corev1.Secret {
+	labels := map[string]string{
+		"app": cr.Spec.AppName,   
+	}
+
+	crt, key := tlstools.GenerateCrt("server")
```

Comment:
How does someone provide their own cert/key if they have them?

Reply:
they'd have to put them somewhere inside this project dir and operator reads from it. this requires code change. now it's tls secrets are randomly generated.

Reply:
It would also require an operator image rebuild, right? That's ... not great. We can tackle this in a followup though.